clientv3/watch.go

```diff
@@ -50,11 +50,13 @@ type Watcher interface {
 	// and "WatchResponse" from this closed channel has zero events and nil "Err()".
 	// If the context is "context.Background/TODO", returned "WatchChan" will not be closed
 	// and wait until events happen, except when server returns a non-recoverable error.
-	// For example, when context passed with "WithRequireLeader" and the connected server
-	// has no leader, error "etcdserver: no leader" is returned, and then "WatchChan" is
-	// closed with non-nil "Err()".
-	// Otherwise, as long as the context has not been canceled or timed out, watch will
-	// retry on other recoverable errors forever until reconnected.
+	// In other words, a large number of watchers with "context.Background/TODO" can
+	// lead to excessive memory usage.
+	// One of the non-recoverable errors from watch API is "etcdserver: no leader"
```

Comment:
let us not change anything else in this PR, just add one simple sentence about the cancelling of the context.